src/routes/Profile/ClaimButton/ClaimButton.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useCallback, useRef, useState } from 'react';
+import {
+  ClaimedAtTypography,
+  ClaimButton as ClaimButtonUI,
+  StretchClaimButton,
+  TickWrapper,
+} from './ClaimButton.styled';
+import { Confetti } from './Confetti/Confetti';
+import { Tick } from './Tick/Tick';
+import { BouncedLoading } from './BouncedLoading/BouncedLoading';
+import { formatPOSIXTimestamp } from '@utilities';
+
+type ClaimButtonMode = 'claim' | 'claimed' | 'claiming' | 'claimedDate';
+
+export type ClaimButtonProps = {
+  mode: ClaimButtonMode;
+  claimedAt?: number;
+  onClick?: () => void;
+};
+
+const MODE_COPY_MAP: Record<ClaimButtonProps['mode'], string> = {
+  claim: 'Claim',
+  claimed: 'Claimed',
+  claiming: 'Claiming',
+  claimedDate: 'Claimed:',
+};
+
+export const ClaimButton: React.FunctionComponent<ClaimButtonProps> = ({
+  onClick,
+  mode,
+  claimedAt,
+}) => {
+  const [showConfetti, setShowConfetti] = useState(false);
+  const initialModeRef = useRef<ClaimButtonMode>(mode);
+  const ButtonUI = mode === 'claimedDate' ? StretchClaimButton : ClaimButtonUI;
+  const Wrapper = showConfetti ? Confetti : React.Fragment;
```

Comment:
it is same as <> </> you are used to... I just want to specify the wrapper around a component dynamically and I don't want to use div since it will affect styling